### PR TITLE
Fixes 1201079 - Stop appending a forward slash to every URL

### PR DESCRIPTION
**Pull request by st3fan**
This patch makes sure that we do not display a trailing slash for an `NSURL` that has an empty (`/`) or `nil` path component.

If the `NSURL` contains a non-empty path that ends with a trailing slash then we do not do anything. This is because that will actually result in a different URL. So for example `http://www.apple.com/iphone/` will not be changed. This is the same as *Firefox Desktop* handles it.

I have merged this logic in `absoluteStringWithoutHTTPScheme`, which is called by the `BrowserLocationView.updateTextWithURL()` to update the currently displayed URL. Because this now does more than just remove the `http://` part, I have renamed that function to `absoluteDisplayString`.

Also included tests to make sure this works on URLs with and without trailing slashes with both empty and non-empty path compoments.

Feels good to write a patch 😎

### CODE REVIEW

File: Utils/Extensions/NSURLExtensions.swift [hunk 1]
```diff
@@ -104,11 +104,16 @@ extension NSURL {
...
+        if var urlString = self.absoluteString {
+            // For http URLs, get rid of the trailing slash if the path is empty or '/'
+            if (self.scheme == "http" || self.scheme == "https") && (self.path == "/" || self.path == nil) && urlString.endsWith("/") {
+                urlString = urlString.substringToIndex(advance(urlString.endIndex, -1))
+            }
             // If it's basic http, strip out the string but leave anything else in
             if urlString.hasPrefix("http://") ?? false {
-                return urlString.substringFromIndex(advance(urlString.startIndex, 7))
+                let urlWithoutSchemeString = urlString.substringFromIndex(advance(urlString.startIndex, 7))
+                return urlWithoutSchemeString
```
Comment (thebnich) [inline Utils/Extensions/NSURLExtensions.swift]: Nit: you could just return the result here directly without creating the extra `urlWithoutSchemeString` variable

File: SharedTests/NSURLExtensionsTests.swift [hunk 2]
```diff
@@ -9,25 +9,61 @@ import Shared
...
         }
     }
 
+    func testKeepsHTTPSAndRemovesTrailingSlashInURL() {
+        let url = NSURL(string: "https://google.com/")
+        if let actual = url?.absoluteDisplayString() {
+            XCTAssertEqual(actual, "https://google.com")
+        } else {
+            XCTFail("Actual url is nil")
+        }
```
Comment (thebnich) [inline SharedTests/NSURLExtensionsTests.swift]: We generally just use force unwrapping in tests, so you could do `XCTAssertEqual(url!.absoluteDisplayString(), "https://google.com")` without any of the `if`/`else` stuff (and it will throw/fail either way). But I guess this is fine too if you prefer it.

File: Utils/Extensions/NSURLExtensions.swift (hunk 1 above)
Comment (st3fan) [inline Utils/Extensions/NSURLExtensions.swift]: Good catch, i'll fix that.
